cdap-watchdog/src/main/java/co/cask/cdap/logging/write/FileMetaDataManager.java

```diff
@@ -87,13 +87,25 @@ public TransactionExecutor createExecutor(Iterable<TransactionAware> txAwares) {
   public void writeMetaData(final LoggingContext loggingContext,
                             final long startTimeMs,
                             final Location location) throws Exception {
+    writeMetaData(loggingContext.getLogPartition(), startTimeMs, location);
+  }
+
+  /**
+   * Persists meta data associated with a log file.
+   * @param logPartition partition name that is used to group log messages
+   * @param startTimeMs start log time associated with the file.
+   * @param location log file.
+   */
+  public void writeMetaData(final String logPartition,
```

Comment:
this seems bad.  Doesn't look like this class intended to expose that the log partition is used in this way... is there any way to get the LoggingContext from the row key?